iso15118/shared/security.py

```diff
@@ -532,9 +533,15 @@ def verify_certs(
     # TODO We also need to check each certificate's attributes for
     #      compliance with the corresponding certificate profile
     for cert in sub_ca_der_certs:
-        path_len = cert.extensions.get_extension_for_oid(
-            ExtensionOID.BASIC_CONSTRAINTS
-        ).value.path_length
+        try:
+            basic_contrains = cert.extensions.get_extension_for_oid(
+                ExtensionOID.BASIC_CONSTRAINTS
+            ).value
+            path_len = 0
+            if isinstance(basic_contrains, extensions.BasicConstraints):
+                path_len = basic_contrains.path_length
+        except ExtensionNotFound:
+            path_len = None
```

Comment:
I think if we set path_len to None, we risk raising another exception while trying to raise CertChainLengthError below. as num_sub_cas is an int. Is it better to modify the definition for that to include another message field and making num_sub_cas optional?

Reply:
Good catch! I think I tried to set it as "0" instead of None, so the next piece of code would raise the "CertAttributeError" exception. Either way, I'll raise that exception directly.